### PR TITLE
feat(core): Add generic shader module exclusion mechanism

### PR DESCRIPTION
**Pull request by akre54**
#### Background
The `mergeShaders()` function had a hardcoded check that removed `project32` when `project64` appeared in the merged modules list. This hack was brittle — any future mutually-exclusive module pair would need another manual check.

#### Change List
- Add `excludes?: string[]` field to ShaderModule interface via module augmentation
- Update project32 and project64 modules to declare `excludes: ['project64']` and `excludes: ['project32']` respectively
- Replace hardcoded check with generic exclusion algorithm in mergeShaders() using "later wins" precedence
- Add comprehensive unit tests covering original behavior, reverse case, generic mechanism, and edge cases

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by akre54** [comment]
This could be added to `@luma.gl/shadertools` directly too

File: modules/core/src/utils/shader.ts [hunk 1]
```diff
@@ -2,6 +2,17 @@
...
 // Copyright (c) vis.gl contributors
 
+declare module '@luma.gl/shadertools' {
+  interface ShaderModule {
+    /**
+     * Names of modules that cannot co-exist with this module.
+     * When multiple mutually exclusive modules are present, the one appearing
+     * later in the merge order takes precedence.
+     */
+    excludes?: string[];
```
Comment (chrisgervang) [inline modules/core/src/utils/shader.ts:12]: Please upstream

**Review by chrisgervang — CHANGES_REQUESTED** [review]
This looks like a change that belongs in luma.

What's the use case?

**Comment by akre54** [comment]
Wanted to get feedback for deck before [proposing on luma](https://github.com/visgl/deck.gl/pull/10714#issuecomment-5708936379)

The use case is this change, since the modules are mutually exclusive
